File: chapter03/src/main/java/com/example/chapter03/part3_valueanimator_advanced_ofobject/CharEvaluator.java
package com.example.chapter03.part3_valueanimator_advanced_ofobject;

import android.animation.TypeEvaluator;

/**
 * @author wangzhichao
 * @since 2021/5/11
 */
public class CharEvaluator implements TypeEvaluator<Character> {
    @Override
    public Character evaluate(float fraction, Character startValue, Character endValue) {
        int startInt = startValue;
        int endInt = endValue;
        return ((char) (startInt + fraction * (endInt - startInt)));
    }
}

File: chapter03/src/main/java/com/example/chapter03/part3_valueanimator_advanced_ofobject/CharEvaluator.kt
package com.example.chapter03.part3_valueanimator_advanced_ofobject

import android.animation.TypeEvaluator

/**
 * @author wangzhichao
 * @since 2021/5/11
 */
class CharEvaluator : TypeEvaluator<Char> {
    override fun evaluate(fraction: Float, startValue: Char, endValue: Char): Char {
        val startInt = startValue.code
        val endInt = endValue.code
        return (startInt + fraction * (endInt - startInt)).toInt().toChar()
    }
}